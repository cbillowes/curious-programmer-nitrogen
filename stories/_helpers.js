import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page