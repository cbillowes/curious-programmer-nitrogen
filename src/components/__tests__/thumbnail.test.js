import React from "react"
import Thumbnail from "../thumbnail"
import { getSnapshot } from "./_helpers"

describe(`Thumbnail`, () => {
  it(`should render with no photo`, () => {
    const component = (
      <Thumbnail
        number={42}
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })

  it(`should render with a specific image`, () => {
    const component = (
      <Thumbnail
        number={42}
        photo="dolphin.jpg"
        credit="Douglas Adams"
        creditLink="https://www.douglas.adams"
        creditSource="personal"
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })

  it(`should render without attribution`, () => {
    const component = (
      <Thumbnail
        number={42}
        photo="dolphin.jpg"
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })
})
