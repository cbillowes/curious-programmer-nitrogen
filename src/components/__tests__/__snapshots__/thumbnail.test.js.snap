// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail should render with a specific image 1`] = `
<div
  className="thumbnail"
>
   
  <a
    className="slug"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        aria-hidden={true}
        style={
          Object {
            "paddingBottom": "NaN%",
            "width": "100%",
          }
        }
      />
      <picture>
        <source />
        <img
          alt=""
          loading="lazy"
          onLoad={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </a>
   
   
  <a
    className="badge linked"
    title="Download free do whatever you want high-resolution photos from Arget"
  >
    <span
      className="source-logo"
    >
      <img
        alt="unsplash"
        src="/unsplash.svg"
      />
    </span>
    <span
      className="attribute"
    >
      Arget
    </span>
  </a>
   
</div>
`;

exports[`Thumbnail should render with no photo 1`] = `
<div
  className="thumbnail"
>
   
  <a
    className="slug"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        aria-hidden={true}
        style={
          Object {
            "paddingBottom": "NaN%",
            "width": "100%",
          }
        }
      />
      <picture>
        <source />
        <img
          alt=""
          loading="lazy"
          onLoad={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </a>
   
   
  <a
    className="badge linked"
    title="Download free do whatever you want high-resolution photos from Dlanor S"
  >
    <span
      className="source-logo"
    >
      <img
        alt="unsplash"
        src="/unsplash.svg"
      />
    </span>
    <span
      className="attribute"
    >
      Dlanor S
    </span>
  </a>
   
</div>
`;

exports[`Thumbnail should render without attribution 1`] = `
<div
  className="thumbnail"
>
   
  <a
    className="slug"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        aria-hidden={true}
        style={
          Object {
            "paddingBottom": "NaN%",
            "width": "100%",
          }
        }
      />
      <picture>
        <source />
        <img
          alt=""
          loading="lazy"
          onLoad={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </a>
   
   
  <a
    className="badge linked"
    title="Download free do whatever you want high-resolution photos from Arget"
  >
    <span
      className="source-logo"
    >
      <img
        alt="unsplash"
        src="/unsplash.svg"
      />
    </span>
    <span
      className="attribute"
    >
      Arget
    </span>
  </a>
   
</div>
`;
